### ISSUE TITLE
Necro War

### ISSUE THREAD

**Issue by TwinNumberOne** [siggame/MegaMinerAI-Game-Ideas#59]
**TLDR**
-
- Build your dungeon to defend yourself
- Mine and collect resources to build your army and dungeon
- Destroy your enemy with your horrific hoard
- Rock paper scissors like gameplay!
- Magic beats slime, physical beats magic, slime beats physical.

**Story**
-
In this game, you are a master necromancer setting out to prepare your undead hordes to rule the land. You find a nice large empty cave to set up shop and just when you get settled in you notice another necromancer has set up just across the field! So you decide to go over and discuss the problem like a rational adult, I'm kidding, you start summing up all known and unknown horrors to go and destroy your opponent, but you will need to materials to make your horrors out of, mana to power them and gold to hire monsters... time to get to work.

**The Game**
-

**Victory**
The destruction of your enemy or 1000 turns where the user with the most monetary value wins
This is calculated based on what the necromancer has stored at the end, and what state the troops they have left are in, with the troops remaining hp being multiplied by their total value (with gold, materials, and mana being worth 1 of these victory points)

***Map:***
a 120x120 map (to make room for the dungeons) with a 20x20 dungeon in the middle of the left and right side, and a map separator down the middle that is the wave portal. The dungeon will then have a 5x1 entrance with the rest of the area being lined with an indestructible wall. The back of the dungeon will have a mineral mine, a gold mine, and a mana pool that generates a certain amount of mana every turn (units can be pushed into this to convert their hp to mana) with all new units spawning out of a dark hole in the back. The area between the dungeons will have a forest that can be harvested for all 3 resources but will have to be carried back to the dungeon when harvested. The dungeon will then also have movable gold and materials piles that store their respective resource with each pool having a cap of 100 units. 

**movement**
- move around normally by walking or whatever you call what slimes do
-  a be carried by a bigger monster (ohh slimes again!) 
- teleported by a floating skull, or by going through the wave portal in the center of the map, to get to the other player's side. which has a charge up time before releasing the monsters in the huge wave, however, if the monsters step back through they appear next to their respective mana pool.

**Resource collection**
- can be collected from the mines, and generated by the mana pool, the more troops on the mines (those that can build or collect resources) the more materials you will collect.
- from harvested trees
- the body's of your dead foes (as a necromancer you waste not want not, you can toss their bodies in the pool to get a small amount of mana or disassemble their bodies as parts for new troops).
- Skulls can gather mana from the enemy and friendly mana pools.

**unites**
(materials/mana/gold) (in a range form 0-8, zero being none)

| Name | Cost | Health |  Move Speed | attack | special | 
|----|----|----|----|----|----|
|**Skeleton**          |1/1/0|low|High|Physical|SR|
|**Archer**             |4/1/1|Low|Low|Physical ranged|R|
|**Demon**           |8/8/0|Medium|High|Physical AOE|RA|
|**Slime**              |5/2/5|High|Medium|Slime|CW|
|**Skull**               |1/8/3|Mana|High|Magic|M|
|**Necromancer**|nope|High|None|Magic/Physical|M|

S - can build and harvest materials
C - carry and protect other units from physical attacks, as well as materials
R - resistant to magic
W - weak to magic
M - can use mana to attack or for other effects
A - small AOE area (like 3 squares in front of it)

_Skeletons_
- you're standard spooky scary skeleton where you start out with 10.
- useful for gathering resources outside the base or swarming an enemy.
- 5 of them can be on one tile, and gain the knockback ability when they do so (1 tile)

_Archer_
- you're standard spooky scary skeleton but he has a built-in bow and can fire a few squares.

_Demon_
- blasts fire blows up areas, is expensive but worth it. (ignores slimes protection)
- Strong to magic
- Also induces knockback (3 tiles)

_Slime_
- your anti range and magic can be used to move guys around, and they can attack from inside it
- They can also carry materials around, quite a bit more than a single skeleton.

_Skull_
- this is really a minor lich used to channel your magical power on the battlefield.
- its health is its mana, and it can store infinite mana but it does use mana to attack.
-  It can shoot lightning (line attack, 5 tiles), fireball (AOE and knockback 3x3 tiles), repair/heal units. or teleport units to a spot another skull is at, or back to the base like the wave portal (costs mana per turn to maintain).

_Necromancer_
- unites spawn from you
- you can cast the same spells as the lich
- you are stuck in a summoning ritual and can't move

**Buildables**
- you are not allowed to block access to your necromancer with the wall, arrowslit or trap door, but you can line the way with traps
- Wall: your basic indestructible wall. cost 5 materials and 1 gold
- arrow slit: like a wall but your units can attack through it, costs 3 materials and 3 gold
- Trap door: only allows your units to move through it costs 2 materials and 8 gold
- spike trap (physical), costs 3 gold and 2 materials (one use, then needs to be cleared costs 2 gold)
- fireball trap (casts fireball one use) costs 1 material, 1 gold and the cost of a fireball spell
- slime pit (traps an enemy mob) costs 3 materials 1 mana and 3 gold, (can be filled costs 2 gold to clear)
- only the traps are destructible (as listed in their descriptions) and if the workload is too much you could make them upgrade walls to arrow slits and trap doors, but still place traps

**Combat**
Slime ignores physical attacks (demon skeleton archer) but is weak to magic. Skull is weak to physical attacks. The Demon is disabled by the slime. 

**Remarks**
If you had to do anything to make it simpler if you could cut out the demon or the archer units. you could also make the dungeon layouts fixed and make the person upgrade the different spots with traps or arrowslits and the like, and or just make it so that the dungeon is fixed and they can put down traps. Oh also while it was never explicitly stated I will say that the enemy can harvest the trees on the other player's field, they will just have to send the unites through during a wave to do it, and then have them run back through after they have harvested. Also, the idea with the resource piles is that the play can better defend their resources against enemy looting or use them to set up traps for the enemy.

**Comment by JeffreyStrahm** [comment]
AKA dungeon keeper

**Comment by JeffreyStrahm** [comment]
Also, I assume arrow slits and trap doors are indestructible
